model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/liquibase/MapLiquibaseConnectionProvider.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.models.map.storage.jpa.liquibase;
+
+import java.sql.Connection;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import liquibase.Liquibase;
+import liquibase.change.ChangeFactory;
+import liquibase.database.Database;
+import liquibase.database.DatabaseFactory;
+import liquibase.database.jvm.JdbcConnection;
+import liquibase.datatype.DataTypeFactory;
+import liquibase.exception.LiquibaseException;
+import liquibase.resource.ClassLoaderResourceAccessor;
+import liquibase.resource.ResourceAccessor;
+import liquibase.sqlgenerator.SqlGeneratorFactory;
+import org.jboss.logging.Logger;
+import org.keycloak.connections.jpa.updater.liquibase.conn.LiquibaseConnectionProvider;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.map.storage.jpa.liquibase.extension.GeneratedColumnSqlGenerator;
+import org.keycloak.models.map.storage.jpa.liquibase.extension.CreateJsonIndexChange;
+import org.keycloak.models.map.storage.jpa.liquibase.extension.CreateJsonIndexGenerator;
+import org.keycloak.models.map.storage.jpa.liquibase.extension.GeneratedColumnChange;
+import org.keycloak.models.map.storage.jpa.liquibase.extension.JsonDataType;
+
+/**
+ * A {@link LiquibaseConnectionProvider} implementation for the map-jpa module. This provider registers the custom {@code Liquibase}
+ * changes and data types that were developed to better support working with data stored as JSON in the database.
+ * </p>
+ * An instance of this provider has to be obtained via {@link KeycloakSession#getProvider(Class, String)} as follows:
+ * <pre>
+ *     LiquibaseConnectionProvider liquibaseProvider = session.getProvider(LiquibaseConnectionProvider.class, "map-jpa");
+ * </pre>
+ *
+ * @author <a href="mailto:sguilhen@redhat.com">Stefan Guilhen</a>
+ */
+public class MapLiquibaseConnectionProvider implements LiquibaseConnectionProvider {
```

Comment:
Is this a Liquibase _connection_ provider or a MapUpdaterProvider?

Reply:
Connection provider.